eqcorrscan/core/lag_calc.py

```diff
@@ -631,6 +637,11 @@ def lag_calc(detections, detect_data, template_names, templates,
                             template_plot.remove(tr)
                     plot_repicked(template=template_plot, picks=event.picks,
```

Comment:
This takes the usual plotting arguments - currently the default is to show the figure.  I think we should keep the default to be as is (showing the figure), however, if plotdir is given, then we should pass a dictionary of kwargs. Something like:
```python
if plotdir:
    if not os.path.isdir(plotdir):
        os.makedirs(plotdir)
    savefile = "{plotdir}/{rid}.png".format(plotdir=plotdir, rid=event.resource_id.id)
    plot_kwags = dict(show=False, save=True, savefile=savefile)
else:
    plot_kwargs = dict(show=True)
plot_repicked(template=template_plot, picks=event.picks, det_stream=plot_stream, **plot_kwargs)
```
This would also mean that matplotlib doesn't have to be imported here, and would make use of the plotting api a bit better.

Reply:
Hi, dear chamberlain.
Thank you for your guide.
I learn something more from your comment and i thanks very much.
I will try again in correct way.
I hope this time everything will be correct.